### DISCUSSION TITLE
Merge Implementation Issue in Merge Sort

### DISCUSSION THREAD

**Discussion by Robertation256** [TheAlgorithms/Python#11434] [General]
I think [popping from the beginning of the list](https://github.com/TheAlgorithms/Python/blob/master/sorts/merge_sort.py#L41) might be very expensive here and can cause the sorting complexity to deteriorate to O(n^2)

**Comment by Mayureshd-18** [comment]
Popping from the front can indeed be expensive for large lists. Bu int his case we are popping from the list which is referenced in the merge function which is called recursively thus it doesn't affect the overall time complexity.
In case we are popping without recursion it can prove to be expensive

**Reply by Robertation256** [reply]
Not sure what you mean by that but if you consider the last merge pass where you merge two sorted sublists of size N/2, popping from the front of the list in this pass alone immediately yields in O(N^2) overhead.
